File: Dominio/Entidades/Nd.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

#nullable disable

namespace Dominio.Entidades
{
    [Table("Nd")]
    public partial class Nd
    {
        public Nd()
        {
            ItensNds = new HashSet<ItensNd>();
        }

        [Key]
        public int Id { get; set; }
        public int? IdShopping { get; set; }
        [Column(TypeName = "date")]
        public DateTime? DataEmissao { get; set; }
        [ForeignKey("instrumento")]
        public int? IdInstrumento { get; set; }
        [ForeignKey("servicoCobranca")]
        public int? IdServicoCobranca { get; set; }
        public double? ValorOriginal { get; set; }
        public double? ValorPrincipal { get; set; }
        public double? ValorSaldo { get; set; }
        [Column(TypeName = "date")]
        public DateTime? Vencimento { get; set; }
        public int? IdLiquidacao { get; set; }
        public int? IdCobranca { get; set; }
        [Column(TypeName = "date")]
        public DateTime? Liquidacao { get; set; }
        [Column(TypeName = "decimal(12, 7)")]
        public decimal? PercentualMulta { get; set; }
        public int? IdTipoJuros { get; set; }
        [Column(TypeName = "decimal(12, 7)")]
        public decimal? PercentualJuros { get; set; }
        [ForeignKey("beneficiario")]
        public int? PessoaBeneficiario { get; set; }
        [ForeignKey("pagador")]
        public int? PessoaPagador { get; set; }
        [StringLength(20)]
        public string NossoNumero { get; set; }
        [StringLength(1)]
        public string DigitoNossoNumero { get; set; }
        [StringLength(50)]
        public string ArquivoRemessa { get; set; }
        [Column(TypeName = "date")]
        public DateTime? GeracaoArquivoRemessa { get; set; }
        [StringLength(50)]
        public string ArquivoRetorno { get; set; }
        [Column(TypeName = "date")]
        public DateTime? TratamentoArquivoRetorno { get; set; }
        public int? IdFormaCriacao { get; set; }
        public int? Remessa { get; set; }
        public int? MesProcessamento { get; set; }
        public int? AnoProcessamento { get; set; }
        public int? IdSerie { get; set; }
        public int? IdInclusao { get; set; }
        public string? CodigoBarras { get; set; }
        public string? LinhaDigitavel { get; set; }
        [StringLength(100)]
        public string Usuario { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? DataInsercao { get; set; }
        [StringLength(100)]
        public string UsuarioAlteracao { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? DataAlteracao { get; set; }
        
        public virtual ServicoCobranca servicoCobranca { get; set; }

        public virtual Pessoa beneficiario { get; set; }

        public virtual Pessoa pagador { get; set; }

        public virtual Instrumento instrumento { get; set; }

        public virtual ICollection<ItensNd> ItensNds { get; set; }
    }
}

File: Web/Site/src/app/Models/ND.ts
import { Luc } from './Luc';
import { Marca } from './Marca';
import { ItensND } from './ItensND'
import { ServicoCobranca } from './ServicoCobranca'
import { Pessoa } from './Pessoa'

export class ND {
  id: number | undefined;
  idShopping: number | undefined;
  idInstrumento: number | undefined;
  idServicoCobranca: number | undefined;
  valorOriginal: number | undefined;
  valorPrincipal: number | undefined;
  valorSaldo: number | undefined;
  vencimento: Date | undefined;
  idLiquidacao: number | undefined;
  idCobranca: number | undefined;
  liquidacao: number | undefined;
  percentualMulta: number | undefined;
  idTipoJuros: number | undefined;
  percentualJuros: number | undefined;
  pessoaBeneficiario: number | undefined;
  pessoaPagador: number | undefined;
  nossoNumero: string | undefined;
  digitoNossoNumero: string | undefined;
  arquivoRemessa: string | undefined;
  geracaoArquivoRemessa: Date | undefined;
  arquivoRetorno: string | undefined;
  tratamentoArquivoRetorno: Date | undefined;
  idFormaCriacao: number | undefined;
  remessa: number | undefined;
  idInclusao: number | undefined;
  usuario: string | undefined;
  dataInsercao: Date | undefined;
  usuarioAlteracao: string | undefined;
  dataAlteracao: Date | undefined;
  mesProcessamento: number | undefined;
  anoProcessamento: number | undefined;
  idSerie: number | undefined;
  nomeSerie: string | undefined;
  luc: Luc;
  marca: Marca;
  itensNd: ItensND[];
  tipoLocacaoFiltro: number | undefined;
  lucsFiltro: any[];
  marcasFiltro: any[];
  vencimentoNDFiltroDe: Date | undefined;
  vencimentoNDFiltroAte: Date | undefined;
  servicoCobranca: ServicoCobranca;
  pagador: Pessoa;
  beneficiario: Pessoa;
  codigoBarras: string;
}
